app/src/main/java/ar/delellis/quicknotes/api/helper/NoteCompare.java

```diff
@@ -0,0 +1,58 @@
+package ar.delellis.quicknotes.api.helper;
+
+import com.google.gson.Gson;
+
+import ar.delellis.quicknotes.model.Note;
+
+/**
+ * Helper to compare different versions of notes for major changes
+ */
+public final class NoteCompare {
+
+    /**
+     * Compare specific fields from two Note elements.
+     * If there is any difference or fields are present in one,
+     * but not the other objects, this will return true.
+     *
+     * @param shadowCopy saved copy Note
+     * @param current    currently edited Note
+     * @return true if there is shadowCopy difference in the relevant fields
+     */
+    public static boolean compareNotes(Note shadowCopy, Note current) {
```

Comment:
Finally you do not need a new class to compare the notes. Just add the comparison method in the model.

https://github.com/matiasdelellis/nextcloud-quicknotes-android/blob/master/app/src/main/java/ar/delellis/quicknotes/model/Note.java

Example. https://github.com/matiasdelellis/nextcloud-quicknotes-android/blob/master/app/src/main/java/ar/delellis/quicknotes/model/Tag.java#L57

In fact, to compare something like this should be enough, since it compares all the elements. Please test it.

Reply:
Depends how you prefer it - Changed it and moved them to the model.